config/http_config.go

```diff
@@ -769,9 +848,13 @@ func (rt *oauth2RoundTripper) RoundTrip(req *http.Request) (*http.Response, erro
 		changed bool
 	)
 
-	if rt.clientSecret != nil {
+	clientSecret, err := secretFrom(rt.opts.secretManager, rt.config.ClientSecret, rt.config.ClientSecretFile, rt.config.ClientSecretRef)
+	if err != nil {
+		return nil, fmt.Errorf("unable to use client secret: %w", err)
+	}
+	if clientSecret != nil {
 		var err error
-		secret, err = rt.clientSecret.fetch()
+		secret, err = clientSecret.fetch(req.Context())
 		if err != nil {
 			return nil, fmt.Errorf("unable to read oauth2 client secret: %w", err)
 		}
```

Comment:
I guess I missed this in prev PR, but this comment is still relevant https://github.com/prometheus/common/pull/572#discussion_r1595453630